src/state.ts

```diff
@@ -0,0 +1,28 @@
+// Inspired by action/checkout state handling
+import { getState, saveState } from '@actions/core';
+
+// Indicates whether the POST action is running
+export const isPost = !!getState('isPost');
+
+export function setTfRequestId(requestId: string): void {
+  saveState('requestId', requestId);
+}
+
+export function getTfRequestId(): string {
+  return getState('requestId');
+}
+
+export function setTfArtifactUrl(requestUrl: string): void {
+  saveState('artifactUrl', requestUrl);
+}
+
+export function getTfArtifactUrl(): string {
+  return getState('artifactUrl');
+}
+
+// Publish a variable so that when the POST action runs, it can determine it should run the cleanup logic.
+// This is necessary since we don't have a separate entry point.
+// ?NOTE: Post is invoked only when cancelled() is true
+if (!isPost) {
```

Comment:
Do I read this correctly to be: if not 'isPost', then save 'isPost' as true? Isn't the conditional reversed?

Reply:
No, please see the comment above.

> Publish a variable so that when the POST action runs, it can determine it should run the cleanup logic.
> This is necessary since we don't have a separate entry point.

So, when `isPost` is `false`, we are not in post, then we set `isPost` to `true` because after it, there will be a post phase (if `cancelled()` would be `true`).